### ISSUE TITLE
Allow running in Wayland sessions by default

### ISSUE THREAD

**Issue by Peque** [flathub/org.musescore.MuseScore#111]
I installed Musescore from Flathub.

Using Flatseal it seems the X11 windowing system socket was available by default while the Wayland windowing system socket was not available by default. Also, `QT_QPA_PLATFORM` was explicitly set to `xcb`. This lead to a lot of glitches in my Wayland desktop (Fedora 39), making Musescore basically unusable.

Once I changed `QT_QPA_PLATFORM` to `wayland` and enabled access to Wayland windowing system everything worked just fine.

Maybe Wayland is nowadays more commonly used, maybe not, maybe that will change in the future so I would not suggest replacing one configuration with the other right now.

However, I noticed that if both X11 and Wayland windowing systems are available *and* `QT_QPA_PLATFORM` is simply removed (not set to `wayland` nor `xcb`), it worked just fine too.

Wouldn't it be possible to remove `QT_QPA_PLATFORM` so that accessing one socket or the other is inferred according to the environment it is running on? Of course, leaving access to both Wayland and X11 for compatibility with all systems?

So intstead of:

- Sockets:
  - Wayland windowing system [off]
  - X11 windowing system [on]
- Variables:
  - `QT_QPA_PLATFORM=xcb`

Use:

- Sockets:
  - Wayland windowing system [on]
  - X11 windowing system [on]
- Variables:
  - [removed]

Or maybe what I suggest is not currently viable for some reason?

**Comment by Peque** [comment]
PS: It seems the Wayland support was removed more than a year ago as "Too many wayland bugs":

https://github.com/flathub/org.musescore.MuseScore/commit/8c4d3af6750034a99b9e38d189d5a407d0026201

The situation has probably improved a lot since then since I don't seem to experience any issues with Wayland. :blush:

**Comment by hfiguiere** [comment]
no it has not. It's still Qt 5.15. And when VST3 finally works, it won't be an option anyway to run Wayland.

**Comment by Peque** [comment]
@hfiguiere When you say "no it has not" are you replying to my "the situation has probably improved" comment or to something else?

If the former, I seem to be running MuseScore perfectly fine with Wayland, so why not make it easier by default for all Wayland users when they install MuseScore from Flathub? (i.e.: let them run the software without having to tweak any parameters with Flatseal)

I assume X11 users would still be happy with the changes I proposed above or would anything break for X11 users?

**Comment by hfiguiere** [comment]
let me rephrase it: Wayland support will NOT be enabled. It doesn't provide an out of the box experience.

**Comment by Peque** [comment]
@hfiguiere Would you mind sharing what is missing from an "out of the box experience"?

**Comment by Peque** [comment]
@hfiguiere Friendly ping. :innocent: 

Note I have been using MuseScore a couple of days from Wayland and everything I use seems to work just fine: opening scores, saving to cloud, exporting MP3, playing, editing... No performance issues and zero crashes so far (much better user experience than the current default of disabled Wayland support).

**Comment by pianocomposer321** [comment]
@hfiguiere I'm a Linux user who daily drives a Wayland desktop environment. I have no issues at all with any application *except* for musescore. Is there any reason not to enable Wayland support? You say that "It doesn't provide an out of the box experience", but that's a very confusing statement considering that the problem right now is that it doesn't work for Wayland users out of the box. Wayland is used by a large number of Linux users. If enabling Wayland support out of the box would not break musescore for x11 users (which OP seems to think is the case), is there a good reason not to?

Your comments and closing of the issue without explanation currently are coming across as very user-hostile.
